src/course.py

```diff
@@ -0,0 +1,92 @@
+import os
+import re
+import urllib.parse
+
+from bs4 import BeautifulSoup
+
+import globals
+from resource import Resource
+
+
+class Course:
+    def __init__(self, course_name, course_url):
+        self.name = course_name
+        self.course_url = course_url
+
+        self.course_page = globals.global_session.get(self.course_url).content
+        self.soup = BeautifulSoup(self.course_page, 'html.parser')
+        self._extract_resources()
+
+    def _extract_resources(self):
+        self.resources = {}
+        self.latest_resources = []
+        sections = []
+        latest_week_section = None
+
+        # Extract sections for courses structured by weeks
+        weeks = self.soup.find('ul', class_='weeks')
+        if weeks is not None:
+            sections += weeks.find_all('li', class_='section main clearfix')  # All week secation
+            latest_week_section = weeks.find('li', class_='section main clearfix current')
+            if latest_week_section is not None:
+                sections.append(latest_week_section)
+
+        # Extract sections for courses structured by topics
+        topics = self.soup.find('ul', class_='topics')
+        if topics is not None:
+            sections += topics.find_all('li', class_='section main clearfix')  # All topic sections
+
+        for section in sections:
+            section_resources = section.find_all('div', class_='activityinstance')
+            for resource_div in section_resources:
+                resource = Resource(resource_div)
+                self.resources[resource.name] = resource
+                if section == latest_week_section:
+                    self.latest_resources.append(resource)
+
+    def download_resource(self, req_name, destination_dir, update_handling):
```

Comment:
Maybe rename `req_name` to `resource_name` or something similar? I think `req_name` isn't totally clear until you read the comment below it.